### PR TITLE
Provide a Draggable that starts on drag

### PR DESCRIPTION
**Pull request by Hixie**
This lets it cooperate with other gestures like tap.

The way I implemented this was to refactor the entire Draggable gesture logic to use a new kind of gesture detector called MultiDragGestureRecognizer. It works a bit like MultiTapGestureRecognizer but for drags.

Also some tweaks to the velocity estimator.

### CODE REVIEW

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/996

File: examples/widgets/drag_and_drop.dart [hunk 1]
```diff
@@ -117,6 +126,14 @@ class ExampleDragSource extends StatelessComponent {
...
         dragAnchor: anchor
       );
+    // } else if (under) {
+    //   return new TapDraggable<Color>(
+    //     data: color,
+    //     child: contents,
+    //     feedback: feedback,
+    //     feedbackOffset: feedbackOffset,
+    //     dragAnchor: anchor
+    //   );
```
Comment (abarth) [inline examples/widgets/drag_and_drop.dart]: Commented out code?

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/gestures/multidrag.dart [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+import 'constants.dart';
+import 'events.dart';
+import 'pointer_router.dart';
+import 'recognizer.dart';
+import 'velocity_tracker.dart';
+
+typedef Drag GestureMultiDragStartCallback(Point position);
+
+class Drag {
+  void move(Offset offset) { }
```
Comment (abarth) [inline packages/flutter/lib/src/gestures/multidrag.dart]: No start with a Position ?

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/gestures/multidrag.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter/lib/src/gestures/multidrag.dart]: You get the position in the onStart, which is where you create the Drag.

**Comment by Hixie** [comment]
I'm checking this in with a known regression: long-press draggable doesn't wait for a long-press if there's no other gesture recognisers there. I'll fix that next.

**Comment by Hixie** [comment]
fix is in https://github.com/flutter/flutter/pull/1790

**Pull request merged**